server/src/main/java/keywhiz/service/permissions/PermissionCheckImpl.java

```diff
@@ -0,0 +1,21 @@
+package keywhiz.service.permissions;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+public class PermissionCheckImpl implements PermissionCheck{
```

Comment:
I don't think you'll actually end up needing this class, it's just a delegate with no additional behavior.